sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -0,0 +1,735 @@
+                            if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, Compilation.GetTypeByMetadataName(Constants.HttpTriggerBindingType)))
+                            {
+                                hasHttpTrigger = true;
+                            }
+                            else if (SymbolEqualityComparer.Default.Equals(attribute.AttributeClass, Compilation.GetTypeByMetadataName(Constants.EventHubsTriggerType)))
+                            {
+                                // there are special rules for EventHubsTriggers that we will have to validate
+                                validEventHubs = IsEventHubsTriggerValid(parameterSymbol, parameter.Type, model, attribute, out dataType, out cardinality);
+                                if (!validEventHubs)
+                                {
+                                    _context.ReportDiagnostic(Diagnostic.Create(DiagnosticDescriptors.InvalidEventHubsTrigger, parameter.Identifier.GetLocation(), nameof(parameterSymbol)));
+                                    bindingsList = null;
+                                    return false;
+                                }
+                            }
+
+                            string bindingName = parameter.Identifier.ValueText;
+
+                            if (!TryCreateBindingDict(attribute, bindingName, parameter.Identifier.GetLocation(), out IDictionary<string, string>? bindingDict))
+                            {
+                                bindingsList = null;
+                                return false;
+                            }
+                            
+                            if (dataType is not DataType.Undefined)
+                            {
+                                bindingDict!.Add("dataType", FormatObject(Enum.GetName(typeof(DataType), dataType)));
+                            }
+
+                            // special handling for EventHubsTrigger - we need to define a property called "Cardinality"
+                            if (validEventHubs)
+                            {
+                                if (!bindingDict!.Keys.Contains("Cardinality"))
+                                {
+                                    bindingDict["Cardinality"] = cardinality is Cardinality.Many ? FormatObject("Many") : FormatObject("One");
+                                }
+                            }
+
+                            bindingsList.Add(bindingDict!);
+                        }
+                    }
+                }
+
+                return true;
+            }
+
+            /// <summary>
+            /// Checks for and returns any bindings found in the Return Type of the method
+            /// </summary>
+            private bool TryGetReturnTypeBindings(MethodDeclarationSyntax method, SemanticModel model, bool hasHttpTrigger, bool hasOutputBinding, out IList<IDictionary<string, string>>? bindingsList)
```

Comment:
Readability nit: This method seems big. See if you can split the code into smaller methods.